kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/JsonClassCodegenProcessor.kt

```diff
@@ -59,6 +59,13 @@ public class JsonClassCodegenProcessor : AbstractProcessor() {
      *   * `"javax.annotation.Generated"` (JRE <9)
      */
     public const val OPTION_GENERATED: String = "moshi.generated"
+
+    /**
+     * This annotation processing argument can disable proguard rule generating.
+     * Normally, this is not recommended unless end-users build their own JsonAdapter look-up tool.
+     * This is enabled by default
+     */
+    public const val OPTION_PROGUARD_CODE_GENERATED: String = "moshi.enabledProguardGenerated"
```

Comment:
Let's call this `OPTION_ENABLE_PROGUARD_RULE_GENERATION` and the value `mocha.enableProguardRuleGeneration`

Reply:
updated on bf273da